avalon/photoshop/lib.py

```diff
@@ -51,359 +54,94 @@ def show(module_name):
     app.exec_()
 
 
-def get_com_objects():
-    """Wrapped com objects.
-
-    This could later query whether the platform is Windows or Mac.
-    """
-    from . import com_objects
-    return com_objects
-
-
-def Dispatch(application):
-    """Wrapped Dispatch function.
+class ConnectionNotEstablishedYet(Exception):
+    pass
 
-    This could later query whether the platform is Windows or Mac.
 
-    Args:
-        application (str): Application to dispatch.
+def stub():
     """
-    from win32com.client import Dispatch
-    return Dispatch(application)
-
-
-def app():
-    """Convenience function to get the Photoshop app.
-
-    This could later query whether the platform is Windows or Mac.
-
-    This needs to be a function call because when calling Dispatch directly
-    from a different thread will result in "CoInitialize has not been called"
-    which can be fixed with pythoncom.CoInitialize(). However even then it will
-    still error with "The application called an interface that was marshalled
-    for a different thread"
+        Convenience function to get server RPC stub to call methods directed
+        for host (Photoshop).
+        It expects already created connection, started from client.
+        Currently created when panel is opened (PS: Window>Extensions>Avalon)
+    :return: <PhotoshopClientStub> where functions could be called from
     """
-    return Dispatch("Photoshop.Application")
+    stub = PhotoshopServerStub()
+    if not stub.client:
+        raise ConnectionNotEstablishedYet("Connection is not created yet")
+
+    return stub
 
 
 def safe_excepthook(*args):
     traceback.print_exception(*args)
 
 
 def launch(application):
-    """Starts the web server that will be hosted in the Photoshop extension.
+    """Starts the websocket server that will be hosted
+       in the Photoshop extension.
     """
     from avalon import api, photoshop
 
     api.install(photoshop)
     sys.excepthook = safe_excepthook
-
-    # Launch Photoshop and the html server.
+    # Launch Photoshop and the websocket server.
     process = subprocess.Popen(application, stdout=subprocess.PIPE)
-    server = html_server.app.start_server(5000)
+
+    websocket_server = WebSocketServer()
+    websocket_server.websocket_thread.start()
 
     while True:
         if process.poll() is not None:
             print("Photoshop process is not alive. Exiting")
-            server.shutdown()
+            websocket_server.close()
             sys.exit(1)
         try:
-            _app = photoshop.app()
-            if _app:
+            _stub = photoshop.stub()
+            if _stub:
                 break
         except Exception:
-            time.sleep(0.1)
+            time.sleep(0.5)
 
     # Wait for application launch to show Workfiles.
     if os.environ.get("AVALON_PHOTOSHOP_WORKFILES_ON_LAUNCH", False):
```

Comment:
name of environment variable could be changed to something shorter like:
`WORKFILES_ON_LAUNCH`

Reply:
That would implicate that is all DCC wide configuration, this seems more granular.